### PR TITLE
Feature/unidirectional sentiment

### PR DESCRIPTION
**Pull request by curt-tigges**

### CODE REVIEW

File: utils/circuit_analysis.py [hunk 1]
```diff
@@ -118,6 +118,68 @@ def get_logit_diff(
 
     return (left_logits - right_logits).mean()
 
+def get_log_probs(
+        logits: Float[Tensor, "batch seq d_vocab"],
```
Comment (ojh31) [inline utils/circuit_analysis.py:122]: Minor: indentation off here

File: utils/circuit_analysis.py [hunk 2]
```diff
@@ -118,6 +118,68 @@ def get_logit_diff(
...
+    # average over the answer tokens
+    answer_log_probs: Float[Tensor, "batch"] = answer_log_probs.mean(dim=1)
+    if per_prompt:
+        return answer_log_probs
+    else:
+        return answer_log_probs.mean()
+
+
+def log_prob_diff_noising(
+        logits: Float[Tensor, "batch seq d_vocab"],
```
Comment (ojh31) [inline utils/circuit_analysis.py:152]: Minor: indentation

File: utils/circuit_analysis.py [hunk 3]
```diff
@@ -118,6 +118,68 @@ def get_logit_diff(
...
+    """
+    log_prob = get_log_probs(logits, answer_tokens)
+    ld = ((log_prob - flipped_log_prob) / (clean_log_prob  - flipped_log_prob))
+    if return_tensor:
+        return ld
+    else:
+        return ld.item()
+    
+def log_prob_diff_denoising(
+        logits: Float[Tensor, "batch seq d_vocab"],
```
Comment (ojh31) [inline utils/circuit_analysis.py:168]: Minor: indentation

File: utils/prompts.py [hunk 4]
```diff
@@ -183,6 +201,71 @@ def get_dataset(
...
+        prompt_type, positive_adjectives, negative_adjectives, neutral_adjectives
+    )
+    prompts_dict = {
+        "positive": model.to_tokens(pos_prompts, prepend_bos=True),
+        "negative": model.to_tokens(neg_prompts, prepend_bos=True),
+        "neutral": model.to_tokens(neutral_prompts, prepend_bos=True)
+    }
+    
+    n_prompts = min([prompts_dict[s].shape[0] for s in dataset_sentiments])
+    print(n_prompts)
```
Comment (ojh31) [inline utils/prompts.py:242]: Minor: remove print?

File: utils/circuit_analysis.py [hunk 5]
```diff
@@ -218,171 +280,6 @@ def patch_head_vector(
     return orig_head_vector
 
 
-def path_patching(
```
Comment (ojh31) [inline utils/circuit_analysis.py:221]: Why are you removing this?

File: Circuit Analysis - Simple Sentiment.py [hunk 6]
```diff
@@ -193,68 +169,46 @@ def get_logit_diff(logits, answer_token_indices, per_prompt=False):
...
-    ' dreadful',' bad',' dull',' depressing',' miserable',' tragic',' nasty',' inferior',' horrific',' terrible',
-    ' ugly',' disgusting',' disastrous',' horrendous',' annoying',' boring',' offensive',' frustrating',' wretched',' dire',
-    ' awful',' unpleasant',' horrible',' mediocre',' disappointing',' inadequate'
-    ]
-
-#negative_adjectives = [' lousy', ' dire', ' bad', ' nasty', ' miserable', ' wretched', ' disgusting', ' ugly', ' disastrous', ' tragic']
-
-len(positive_adjectives), len(negative_adjectives)
-
+from utils.prompts import get_onesided_datasets
```
Comment (ojh31) [inline Circuit Analysis - Simple Sentiment.py:180]: Great to see this sort of thing moving to utils!

**Review by ojh31 — APPROVED** [review]
A few minor questions and comments but broadly looks good

File: utils/circuit_analysis.py (hunk 5 above)
Comment (curt-tigges) [reply utils/circuit_analysis.py:221]: This is old path patching stuff that was in the file when I copied it from another repo; for our experiments I'm only using path_patching.py functions.

**Pull request merged**